pkg/scaling/scale_handler.go

```diff
@@ -250,14 +256,71 @@ func (h *scaleHandler) isScaledObjectActive(ctx context.Context, scalers []scale
 
 func (h *scaleHandler) isScaledJobActive(ctx context.Context, scalers []scalers.Scaler, scaledJob *kedav1alpha1.ScaledJob) (bool, int64, int64) {
 	var queueLength int64
-	var targetAverageValue int64
 	var maxValue int64
 	isActive := false
 
+	scalersMetrics := h.getScalersMetrics(ctx, scalers, scaledJob)
+	switch scaledJob.Spec.ScalingStrategy.MultipleScalersOption {
+	case "min":
+		for _, metrics := range scalersMetrics {
+			if (queueLength == 0 || metrics.queueLength < queueLength) && metrics.isActive {
+				queueLength = metrics.queueLength
+				maxValue = metrics.maxValue
+				isActive = metrics.isActive
+			}
+		}
+	case "avg":
+		queueLengthSum := int64(0)
+		maxValueSum := int64(0)
+		length := 0
+		for _, metrics := range scalersMetrics {
+			if metrics.isActive {
+				queueLengthSum += metrics.queueLength
+				maxValueSum += metrics.maxValue
+				isActive = metrics.isActive
+				length++
+			}
+		}
+		if length != 0 {
+			queueLength = divideWithCeil(queueLengthSum, int64(length))
+			maxValue = divideWithCeil(maxValueSum, int64(length))
+		}
+	case "sum":
+		for _, metrics := range scalersMetrics {
+			if metrics.isActive {
+				queueLength += metrics.queueLength
+				maxValue += metrics.maxValue
+				isActive = metrics.isActive
+			}
+		}
+	default: // max
+		for _, metrics := range scalersMetrics {
+			if metrics.queueLength > queueLength && metrics.isActive {
+				queueLength = metrics.queueLength
+				maxValue = metrics.maxValue
+				isActive = metrics.isActive
+			}
+		}
+	}
```

Comment:
Could you please move this to a separate file, for example to package `pkg/scaling/scaledjob/`

Reply:
I refactored the logic and moved the package.